src/mower_logic/src/mower_logic/behaviors/Behavior.h

```diff
@@ -103,11 +105,34 @@ class Behavior {
     enter();
   }
 
+  template <typename ActionSpec>
+  actionlib::SimpleClientGoalState waitForResultOrAborted(
+      actionlib::SimpleActionClient<ActionSpec>* client, const typename ActionSpec::_action_goal_type::_goal_type& goal,
+      double poll_rate = 10) {
+    ros::Rate rate(poll_rate);
+    client->sendGoal(goal);
+
+    while (true) {
+      rate.sleep();
+      if (aborted) {
+        client->cancelGoal();
+        return client->getState();
+      }
+
+      switch (client->getState().state_) {
+        case actionlib::SimpleClientGoalState::ACTIVE:
+        case actionlib::SimpleClientGoalState::PENDING: break;
+        default: return client->getState();
+      }
```

Comment:
Did something like this except kept the double return.

Reply:
Fine with that, but I'm a bit worried that it will return some ACTIVE state, because `getState()` would be called before it even decides to cancel the goal.

Reply:
It's just a race condition, it will just give us the same behavior as if the abort was a few microseconds later.  I think it's easier to just separate the checking for abort vs checking for failed goal, since they should be handled differently.